pkg/probe/http/http.go

```diff
@@ -113,8 +115,11 @@ func DoHTTPProbe(req *http.Request, client GetHTTPInterface) (probe.Result, stri
 			klog.V(4).Infof("Probe terminated redirects for %s, Response: %v", url.String(), *res)
 			return probe.Warning, fmt.Sprintf("Probe terminated redirects, Response body: %v", body), nil
 		}
+		if limitReached {
```

Comment:
```suggestion
		if err == utilio.ErrLimitReached {
```

This is simpler, but at the cost that it be shadowed later